app/Platform/Services/GameSuggestions/Strategies/CommonPlayersStrategy.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Services\GameSuggestions\Strategies;
+
+use App\Models\Game;
+use App\Models\PlayerGame;
+use App\Models\User;
+use App\Platform\Data\GameSuggestionContextData;
+use App\Platform\Enums\GameSuggestionReason;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+
+class CommonPlayersStrategy implements GameSuggestionStrategy
+{
+    public function __construct(
+        private readonly User $user,
+        private readonly Game $sourceGame,
+    ) {
+    }
+
+    public function select(): ?Game
+    {
+        // Get a pseudorandom sample of users who mastered the source game.
+        // Note that this random sample will always be the same, we're sampling using a modulo operator.
+        $masterUserIds = $this->getMasterUserIds();
+
+        if ($masterUserIds->isEmpty()) {
+            return null;
+        }
+
+        // Find most common games mastered by these users.
+        $gameId = PlayerGame::select('game_id')
+            ->whereIn('user_id', $masterUserIds)
+            ->whereAllAchievementsUnlocked()
+            ->where('achievements_total', '>', 0)
+            ->where('game_id', '!=', $this->sourceGame->id)
+            ->whereNull('deleted_at')
+            ->groupBy('game_id')
+            ->orderByRaw('COUNT(*) DESC')
+            ->limit(10)
+            ->value('game_id');
+
+        return $gameId ? Game::find($gameId) : null;
+    }
+
+    public function reason(): GameSuggestionReason
+    {
+        return GameSuggestionReason::CommonPlayers;
```

Comment:
Missing relatedGameContext for "all games suggestion" results

![image](https://github.com/user-attachments/assets/06832ee3-bd88-409f-9068-63277cd60455)